src/components/DatePicker/Common.tsx

```diff
@@ -0,0 +1,313 @@
+import styled from "styled-components";
+import { InputElement, InputWrapper } from "../Input/InputWrapper";
+import { ReactNode, useCallback, useId } from "react";
+import { Icon } from "../Icon/Icon";
+import { Container } from "../Container/Container";
+import { useCalendar, UseCalendarOptions } from "@h6s/calendar";
+import { IconButton } from "../IconButton/IconButton";
+
+const locale = "en-US";
+const selectedDateFormatter = new Intl.DateTimeFormat(locale, {
+  day: "2-digit",
+  month: "short",
+  year: "numeric",
+});
+
+const explicitWidth = "250px";
+
+const HighlightedInputWrapper = styled(InputWrapper)<{ $isActive: boolean }>`
+  ${({ $isActive, theme }) => {
+    return `border: ${theme.click.datePicker.dateOption.stroke} solid ${
+      $isActive
+        ? theme.click.datePicker.dateOption.color.stroke.active
+        : theme.click.field.color.stroke.default
+    };`;
+  }}
+
+  width: ${explicitWidth};
+}`;
+
+interface DatePickerInputProps {
+  isActive: boolean;
+  disabled: boolean;
+  id?: string;
+  placeholder?: string;
+  selectedDate?: Date;
+}
+
+export const DatePickerInput = ({
+  isActive,
+  disabled,
+  id,
+  placeholder,
+  selectedDate,
+}: DatePickerInputProps) => {
+  const defaultId = useId();
+  const formattedSelectedDate =
+    selectedDate instanceof Date ? selectedDateFormatter.format(selectedDate) : "";
+
+  return (
+    <HighlightedInputWrapper
+      $isActive={isActive}
+      disabled={disabled}
+      id={id ?? defaultId}
+    >
+      <Icon name="calendar" />
+      <InputElement
+        data-testid="datepicker-input"
+        placeholder={placeholder}
+        readOnly
+        value={formattedSelectedDate}
+      />
+    </HighlightedInputWrapper>
+  );
+};
+
+// This is taken from InputElement from InputWrapper.tsx
+// We need to be able to change the color of some of the text of the 'input element'
+// but that isn't possible with an input element, so we have to fake it. Thus the name.
+const FakeInputElement = styled.div`
```

Comment:
I believe we can use this to avoid creating a fake input: https://styled-components.com/docs/api#as-polymorphic-prop

Reply:
Nice, much better!